D4Companion.Services/ScreenProcessHandler.cs

```diff
@@ -130,867 +113,514 @@ private void initImageList()
+        private static (double Similarity, Point Locaiton) Find(Image<Gray, byte> image, Image<Gray, byte> target)
         {
-            //_logger.LogDebug($"{MethodBase.GetCurrentMethod()?.Name}");
+            var minVal = 0.0;
+            var maxVal = 0.0;
+            var minLoc = new Point();
+            var maxLoc = new Point();
+            var mat = new Mat();
 
-            var watch = System.Diagnostics.Stopwatch.StartNew();
+            CvInvoke.MatchTemplate(image, target, mat, TemplateMatchingType.SqdiffNormed);
+            CvInvoke.MinMaxLoc(mat, ref minVal, ref maxVal, ref minLoc, ref maxLoc);
 
-            var currentScreen = _currentScreen.Clone();
+            return (minVal, minLoc);
+        }
 
-            // Convert the image to grayscale
-            Image<Gray, byte> currentScreenFilter = new Image<Gray, byte>(currentScreen.Width, currentScreen.Height, new Gray(0));
-            currentScreenFilter = currentScreen.Convert<Gray, byte>();
-            //currentScreenFilter = currentScreenFilter.ThresholdBinaryInv(new Gray(_settingsManager.Settings.ThresholdMin), new Gray(_settingsManager.Settings.ThresholdMax));
+        [LogTime]
+        private bool FindTooltips(Bitmap currentScreen)
+        {
+            var currentScreenImg = currentScreen.ToImage<Bgr, byte>();
+            var currentScreenGray = currentScreenImg.Convert<Gray, byte>();
+            var itemTooltipBag = new ConcurrentBag<ItemTooltipDescriptor>();
 
-            ConcurrentBag<ItemTooltipDescriptor> itemTooltipBag = new ConcurrentBag<ItemTooltipDescriptor>();
             Parallel.ForEach(_imageListItemTooltips.Keys, itemTooltip =>
             {
-                try
-                {
-                    itemTooltipBag.Add(FindTooltip(currentScreenFilter, itemTooltip));
-                }
-                catch (Exception exception)
-                {
-                    _logger.LogError(exception, $"{MethodBase.GetCurrentMethod()?.Name}");
-                }
+                var toolTip = FindTooltip(currentScreenGray, itemTooltip);
+                
+                if (toolTip != null) itemTooltipBag.Add(toolTip);
             });
 
-            // Sort results by similarity
-            var itemTooltips = itemTooltipBag.ToList();
-            itemTooltips.Sort((x, y) =>
-            {
-                return x.Similarity < y.Similarity ? -1 : x.Similarity > y.Similarity ? 1 : 0;
-            });
+            var foundToolTip = default(ItemTooltipDescriptor?);
 
-            foreach (var itemTooltip in itemTooltips)
+            foreach (var itemTooltip in itemTooltipBag)
             {
-                if (itemTooltip.Location.IsEmpty) continue;
+                if (foundToolTip == null || foundToolTip.Similarity < itemTooltip.Similarity)
+                    foundToolTip = itemTooltip;
+            }
 
-                _currentTooltip.Location = itemTooltip.Location;
+            var processedScreen = currentScreen;
```

Comment:
I think something went wrong here? Variable is not being used. Maybe copy/paste issue of the wrong variable.
Looking at your other changes (e.g. `private bool FindItemTypes()`) the logic is also different.